pkg/statistics/builder.go

```diff
@@ -454,8 +466,7 @@ func BuildHistAndTopN(
 //
 //	We assume that the ones not in the top-n list's selectivity is 1/remained_ndv which is the internal implementation of EqualRowCount
 func pruneTopNItem(topns []TopNMeta, ndv, nullCount, sampleRows, totalRows int64) []TopNMeta {
-	// If the sampleRows holds all rows, or NDV of samples equals to actual NDV, we just return the TopN directly.
-	if sampleRows == totalRows || totalRows <= 1 || int64(len(topns)) >= ndv || len(topns) == 0 {
```

Comment:
@time-and-fate I think you need to take a look at this line's change

Reply:
This looks good to me.
I checked the previous bad case and I believe the `int64(len(topns)) >= ndv` check should prevent that.
As for the change from `len(topns) == 0` to `len(topns) <= 1`, it looks like this won't bring a big behavior change.

Reply:
But we may still need to be ready for possible regressions reported from the daily benchmarks or release tests.